sensor/kubernetes/listener/resources/registrymirrorsets.go

```diff
@@ -33,19 +37,52 @@ func (r *registryMirrorDispatcher) ProcessEvent(obj, _ interface{}, action centr
 }
 
 func (r *registryMirrorDispatcher) handleImageContentSourcePolicy(icsp *operatorV1Alpha1.ImageContentSourcePolicy, action central.ResourceAction) *component.ResourceEvent {
-	// TODO(ROX-18251 & ROX-18248): will be implemented as part of the referenced stories.
-	log.Debugf("Received registry mirror ImageContentSourcePolicy event [%v]: %#v", action.String(), icsp)
+	if icsp.GetUID() == "" {
+		log.Warnf("Ignoring ImageContentSourcePolicy - missing UID: %q", icsp.GetName())
+		return nil
+	}
```

Comment:
(applies to the other `GetUID() == ""` checks): is this one required here or is it safe to assume that the object passed to `ProcessEvent` will have the `UID` set? Doesn't seem like the other dispatchers handle this, so curious whether this is specific to these objects or not.

Reply:
An old habit - writing defensively for things that are outside our control. You're right, the other dispatchers do not perform the check, since ACS seems confident that value will always be populated - I'll remove the checks.